### DISCUSSION TITLE
clashing atoms in the conformers from function EmbedMultipleConfs

### DISCUSSION THREAD

**Discussion by LiuCMU** [rdkit/rdkit#7089] [Q&A] [answered]
Hello, I noticed that there are clashing atoms after embedding 3D structures for the following molecule. Here is the reproducing code:

```python
s6 = 'CN1C2=C(C=CC=C2)C#CC3=C(C=CC=C3)C1=O'
mol = Chem.MolFromSmiles(s6)
```
My molecule might be weird, but it exists and is an important reactant (see [paper](https://pubs.acs.org/doi/10.1021/ja3000936) if interested). 
![image](https://github.com/rdkit/rdkit/assets/60156077/bc274771-c31d-4702-9e1c-332bd4483ea4)

I tried to generate an initial conformer and save it to SDF file. The SDF file was empty because 2 atoms clashing together. I verify the clashing atoms by saving the atoms in an xyz file.
```python
mol = Chem.AddHs(mol)
AllChem.EmbedMultipleConfs(mol, numConfs=1, randomSeed=42, numThreads=4, pruneRmsThresh=0.3,
                           useSmallRingTorsions=True, useMacrocycleTorsions=True)
n_confs = mol.GetNumConformers()  # 1
print(n_confs)

writer = Chem.SDWriter('6.sdf')
writer.write(mol, confId=0)  # '6.SDF' is empty due to invalid atomic coordinates

# write the conformer atom positions into XYZ for visualization
conf = mol.GetConformer(0)
with open('6.xyz', 'w') as f:
    f.write(f'29 \n\n')
    for atom in mol.GetAtoms():
        symbol = atom.GetSymbol()
        pos = conf.GetAtomPosition(atom.GetIdx())
        line = symbol + ' ' + str(pos.x)+ ' ' + str(pos.y)+ ' ' + str(pos.z) + '\n'
        f.write(line)
```
Though I understand that the carbon-carbon triple bond is quite strained in this molecule, is there any way to avoid clashing atoms in a conformer during the generation process? Thank you very much!

**Comment by greglandrum** [comment, marked as answer]
@LiuCMU What do you mean that the SDF was empty due to clashing atoms? I can't reproduce that.
Did you see any warnings or error messages?

**Reply by LiuCMU** [reply]
Hello @greglandrum thanks for your follow up!

When I runt the above code,  I only got the `6.xyz`file, and the `6.sdf` file is empty (I checked with a text editor). When visualizing the xyz file, I found clashing atoms (2 carbons are very close to each other). I suspect that the sdf file was empty due to the clashing atoms making the molecule object invalid? 

My rdkit version is 2023.02.3

**Comment by greglandrum** [comment]
> When I runt the above code, I only got the `6.xyz`file, and the `6.sdf` file is empty (I checked with a text editor). When visualizing the xyz file, I found clashing atoms (2 carbons are very close to each other). I suspect that the sdf file was empty due to the clashing atoms making the molecule object invalid?
> 

I don't believe there is any such check in the code. Something else is going on.
If you were running the code interactively (e.g. in a notebook), then the writer won't necessarily write anything to disk unless you either close it or call `flush()`:
```

In [9]: sdw.write(m)

In [10]: !cat foo.sdf

In [11]: sdw.flush()

In [12]: !cat foo.sdf

     RDKit          2D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2990    0.7500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.5981   -0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
M  END
$$$$
```

> My rdkit version is 2023.02.3

There is no such RDKit version. Do you mean 2023.03.2?

**Reply by LiuCMU** [reply]
Thank you so much, indeed I was running the code via a notebook.

My RDKit was installed via conda:
```bash
(py39) tmpry$mamba list rdkit
# packages in environment at /Users/liu5/miniforge3/envs/py39:
#
# Name                    Version                   Build  Channel
rdkit                     2023.03.3        py39h40f44b6_1    conda-forge
```
Please let me know if more info is needed.
